### DISCUSSION TITLE
Function signatures aren't currently editable from within datatype editors, why?

### DISCUSSION THREAD

**Discussion by Wall-AF** [NationalSecurityAgency/ghidra#4541] [Q&A]
Is there a valid reason for the classes [DataTypeEditorManager](https://github.com/NationalSecurityAgency/ghidra/blob/bfb0ed6e34b9c49fc73f4b4cc3db04e49add184d/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/datamgr/editor/DataTypeEditorManager.java#L89), [CompositeEditorModel](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/compositeeditor/CompositeEditorModel.java#L944) and [EditComponentAction](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/compositeeditor/EditComponentAction.java#L56) to exclude the ability to edit a Function Signature, when just adding ` || (baseDt instanceof FunctionDefinition)` (Variable name difers) to the lines indicated in the links would allow it?

**Comment by astrelsky** [comment]
> Is there a valid reason for the classes [DataTypeEditorManager](https://github.com/NationalSecurityAgency/ghidra/blob/bfb0ed6e34b9c49fc73f4b4cc3db04e49add184d/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/datamgr/editor/DataTypeEditorManager.java#L89), [CompositeEditorModel](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/compositeeditor/CompositeEditorModel.java#L944) and [EditComponentAction](https://github.com/NationalSecurityAgency/ghidra/blob/master/Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/compositeeditor/EditComponentAction.java#L56) to exclude the ability to edit a Function Signature, when just adding ` || (baseDt instanceof FunctionDefinition)` (Variable name difers) to the lines indicated in the links would allow it?

If adding those changes works as expected, I'd say to just open pull request. I've been annoyed by this myself a few times.

**Comment by ghidra1** [comment]
The ability to launch this same editor from within the Structure Editor sounds possible, although any such edits would likely remain local to the Structure Editor until its edits are applied.  It would need to be model to the Structure Editor as well.  This is neccessary since the Structure Editor has its own DataTypeManager.

@Wall-AF I can try your suggested changes to see if that works well.

**Reply by Wall-AF** [reply]
They work in my repo.
